src/transaction.cpp

```diff
@@ -880,7 +878,26 @@ bool wsrep::transaction::bf_abort(
 
     if (ret)
     {
+        if (is_streaming() &&
+            client_state_.mode() == wsrep::client_state::m_high_priority)
+        {
+            lock.unlock();
```

Comment:
Why we need to unlock here?

Reply:
These lines were moved from [here](https://github.com/percona/wsrep-lib/pull/3/files/e63647db5b13ca075266ffaf9c19c9bdabd73bf5#diff-93f646c90be7c5557d4795a1d843d047L914) and were added in the below commit to avoid potential deadlocks with the server_state and client_state and other locks related to services.
```
commit 7c6ee3f61fec6bc7ec3537274b18f2324c83d392
Author: Teemu Ollakka <teemu.ollakka@galeracluster.com>
Date:   Mon Oct 15 12:44:04 2018

    In order to avoid potential deadlocks, release client_state lock when
    calling server state methods which may acquire server_state mutex.
    
    Fixed compilation errors in release mode.
```